File: translated/f-to-c/DRB023-sections1-orig-yes.c
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.

SPDX-License-Identifier: (BSD-3-Clause)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//Two tasks without synchronization to protect data write, causing data races.
//Data race pair: i@20:5:W vs. i@22:5:W

#include <omp.h>
#include <stdio.h>

int main()
{
    int i;
    i = 0;

    #pragma omp parallel sections
    {
        #pragma omp section
        i = 1;
        #pragma omp section
        i = 2;
    }

    printf("i = %d\n", i);

    return 0;
} 

File: translated/f-to-go-automodel-bad/DRB023-sections1-orig-yes.go
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
// and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
//
// SPDX-License-Identifier: (BSD-3-Clause)
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

//Two tasks without synchronization to protect data write, causing data races.
//Data race pair: i@20:5:W vs. i@22:5:W

package main

import (
	"fmt"
	"sync"
)

func main() {
	var i int
	i = 0

	//$omp parallel sections
	var wg sync.WaitGroup
	wg.Add(2)

	//$omp section
	go func() {
		defer wg.Done()
		i = 1
	}()
	//$omp section
	go func() {
		defer wg.Done()
		i = 2
	}()
	//$omp end parallel sections

	wg.Wait()

	fmt.Printf("i=%3d\n", i)
}
